splunk_eventgen/lib/generatorplugin.py

```diff
@@ -37,6 +39,66 @@ def __setstate__(self, d):
         self.__dict__ = d
         self._setup_logging()
 
+    def build_events(self, eventsDict, startTime, earliest, latest):
+        eventcount = 0
+        for targetevent in eventsDict:
+            try:
+                event = targetevent['_raw']
+                if event == "\n":
+                    continue
+                # Maintain state for every token in a given event, Hash contains keys for each file name which is
+                # assigned a list of values picked from a random line in that file
+                mvhash = {}
+                pivot_timestamp = EventgenTimestamp.get_random_timestamp(earliest, latest, self._sample.earliest,
+                                                                         self._sample.latest)
+                ## Iterate tokens
+                for token in self._sample.tokens:
+                    token.mvhash = mvhash
+                    event = token.replace(event, et=earliest, lt=latest, s=self._sample,
+                                          pivot_timestamp=pivot_timestamp)
+                    if token.replacementType == 'timestamp' and self._sample.timeField != '_raw':
+                        self._sample.timestamp = None
+                        token.replace(targetevent[self._sample.timeField], et=self._sample.earliestTime(),
+                                      lt=self._sample.latestTime(), s=self._sample, pivot_timestamp=pivot_timestamp)
+                if (self._sample.hostToken):
+                    # clear the host mvhash every time, because we need to re-randomize it
+                    self._sample.hostToken.mvhash = {}
+
+                host = targetevent['host']
+                if (self._sample.hostToken):
+                    host = self._sample.hostToken.replace(host, s=self._sample)
+
+                try:
+                    time_val = int(time.mktime(pivot_timestamp.timetuple()))
+                except Exception:
+                    time_val = int(time.mktime(self._sample.now().timetuple()))
+
+                l = [{'_raw': event,
+                      'index': targetevent['index'],
+                      'host': host,
+                      'hostRegex': self._sample.hostRegex,
+                      'source': targetevent['source'],
+                      'sourcetype': targetevent['sourcetype'],
+                      '_time': time_val}]
+                eventcount += 1
+                self._out.bulksend(l)
+                self._sample.timestamp = None
+            except Exception as e:
+                self.logger.exception("Exception {} happened.".format(type(e)))
+                raise e
+
+        try:
+            endTime = datetime.datetime.now()
```

Comment:
Add a #TODO here about making the outputter no lose all the generated events or to put in a try except break.